### PR TITLE
Fix Issue 15420 - topN(Range, Range) does not respect its less predicate

### PR DESCRIPTION
**Pull request by Infiltrator**

### CODE REVIEW

**Comment by Infiltrator** [comment]
Can I squash using the web interface or do I have to do it myself?

**Comment by JackStouffer** [comment]
You have to do it yourself. See [this](http://wiki.dlang.org/Starting_as_a_Contributor#Someone_asked_me_to_squash_my_commits.2C_what_does_that_mean.3F) section on the wiki.

File: std/algorithm/sorting.d [hunk 1]
```diff
@@ -2308,6 +2308,13 @@ unittest
     topN(a, b);
     sort(a);
     assert(a == [0, 1, 2, 2, 3]);
+
+    // bug 15420
```
Comment (JackStouffer) [inline std/algorithm/sorting.d]: Bug tests should be in their own `unittest` block and should not show up in the docs.

Comment (Infiltrator) [inline std/algorithm/sorting.d]: Well, that explains the /// before the block.  Sorry, I'm not too familiar with ddoc.

**Comment by JackStouffer** [comment]
LGTM

Just for future reference, no one gets a notification when you push your changes, you have to make a comment about it to let people know. I just happened to see this when browsing the PRs.

**Comment by quickfur** [comment]
LGTM

**Comment by quickfur** [comment]
Auto-merge toggled on

**Comment by JackStouffer** [comment]
Hold up @quickfur, just noticed the `{` on the same line.

**Comment by quickfur** [comment]
Auto-merge toggled off

File: std/algorithm/sorting.d [hunk 2]
```diff
@@ -2326,6 +2326,15 @@ unittest
     assert(t == [ 0, 1, 2, 2, 3 ]);
 }
 
+// bug 15420
+unittest {
```
Comment (quickfur) [inline std/algorithm/sorting.d]: Nitpick: opening brace on its own line please.

**Comment by Infiltrator** [comment]
Sorry, force of habit.  I think that I've got my dir-locals configured right for phobos now, though.

**Comment by JackStouffer** [comment]
Ok, now it's good to go ;)

**Comment by JakobOvrum** [comment]
Auto-merge toggled on

**Pull request merged**